src/main/java/ca/bc/gov/backendstartapi/util/ObjectUtil.java

```diff
@@ -0,0 +1,47 @@
+package ca.bc.gov.backendstartapi.util;
+
+import ca.bc.gov.backendstartapi.exception.EmptyObjectNotSupportedException;
+import java.lang.reflect.Array;
+import java.util.Collection;
+import java.util.Objects;
+
+/** This class contains utils methods. */
+public class ObjectUtil {
+
+  private ObjectUtil() {}
+
+  /**
+   * Check if a given object instance is null or empty.
+   *
+   * @param obj Object instance to be checked
+   * @return true if empty or null, false otherwise
+   */
+  public static boolean isEmptyOrNull(Object obj) {
```

Comment:
We must document this method, specifying what types it can receive and what does *empty* mean for each of them: this, however, would leave us with a very convoluted documentation, so I suggest splitting it in different methods (one for `Integer`, one for `String`, etc.) and document each one of them.

A method for `Empty` seems unnecessary, since we can just call `isEmpty` from the object in question.

Reply:
Good catch about documentation. I'll improve it.

I think that having a common class and method for that is more likely for developers to remember. If we have one method for each type, when coding, we will need to check, try to complete to see if exists a method for that type. And having a single method can make things simpler.